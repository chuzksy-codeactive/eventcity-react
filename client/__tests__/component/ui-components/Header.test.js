import React from 'react'
import Connected, { Header } from '../../../src/components/ui-components/Header'

describe('Connected Component', () => {
  it('should render the connected component with all props', () => {
    const store = mockStore({ 
      user: {user: { data: [] }}, 
      auth: { isAdmin: false }, 
      logout: jest.fn() 
    });
    const wrapper = shallow(<Connected store={store} />);
    expect(wrapper.length).toEqual(1);
  });
});

describe('Header Component test', () => {
  const props = {
    user: {
      user: { data: []}
    },
    auth: {
      isAdmin: false
    },
    logout: jest.fn()
  }
  const wrapper = shallow(<Header {...props}/>)
  it('It should render the component without error', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('It should render the component without error', () => {
    wrapper.setProps({
      ...wrapper.props, 
      auth: {
        isAdmin: true
      }
    })
    expect(wrapper.find('#add-center').length).toEqual(1);
  });
})