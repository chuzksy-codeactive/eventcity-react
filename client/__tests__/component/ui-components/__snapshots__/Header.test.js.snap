// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component test It should render the component without error 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    auth={
      Object {
        "isAdmin": false,
      }
    }
    logout={[MockFunction]}
    user={
      Object {
        "user": Object {
          "data": Array [],
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          EventCity
        </a>
        <button
          aria-controls="navbarCollapse"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler navbar-toggler-right"
          data-target="#navbarCollapse"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarCollapse"
        >
          <ul
            className="navbar-nav ml-auto"
          >
            <li
              className="nav-item active"
            >
              <Link
                className="nav-link active"
                exact="true"
                replace={false}
                to="/centers"
              >
                Explore Our Centers
              </Link>
            </li>
            <li
              className="nav-item"
            >
              <Link
                className="nav-link active"
                replace={false}
                to="/login"
              >
                Sign In
              </Link>
            </li>
            <li
              className="nav-item dropdown"
              id="username_link"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                className="nav-link dropdown-toggle"
                data-toggle="dropdown"
                href="/"
                id="navbarDropdownMenuLink"
              />
              <div
                aria-labelledby="navbarDropdownMenuLink"
                className="dropdown-menu"
                id="username_sign_out"
              >
                <Link
                  className="dropdown-item"
                  onClick={[MockFunction]}
                  replace={false}
                  to="/login"
                >
                  Sign Out
                </Link>
              </div>
            </li>
          </ul>
        </div>
      </div>,
      "className": "navbar navbar-expand-md navbar-dark fixed-top bg-dark",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <a
            className="navbar-brand"
            href="/"
          >
            EventCity
          </a>,
          <button
            aria-controls="navbarCollapse"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler navbar-toggler-right"
            data-target="#navbarCollapse"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>,
          <div
            className="collapse navbar-collapse"
            id="navbarCollapse"
          >
            <ul
              className="navbar-nav ml-auto"
            >
              <li
                className="nav-item active"
              >
                <Link
                  className="nav-link active"
                  exact="true"
                  replace={false}
                  to="/centers"
                >
                  Explore Our Centers
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link active"
                  replace={false}
                  to="/login"
                >
                  Sign In
                </Link>
              </li>
              <li
                className="nav-item dropdown"
                id="username_link"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  className="nav-link dropdown-toggle"
                  data-toggle="dropdown"
                  href="/"
                  id="navbarDropdownMenuLink"
                />
                <div
                  aria-labelledby="navbarDropdownMenuLink"
                  className="dropdown-menu"
                  id="username_sign_out"
                >
                  <Link
                    className="dropdown-item"
                    onClick={[MockFunction]}
                    replace={false}
                    to="/login"
                  >
                    Sign Out
                  </Link>
                </div>
              </li>
            </ul>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "EventCity",
            "className": "navbar-brand",
            "href": "/",
          },
          "ref": null,
          "rendered": "EventCity",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-controls": "navbarCollapse",
            "aria-expanded": "false",
            "aria-label": "Toggle navigation",
            "children": <span
              className="navbar-toggler-icon"
            />,
            "className": "navbar-toggler navbar-toggler-right",
            "data-target": "#navbarCollapse",
            "data-toggle": "collapse",
            "type": "button",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "navbar-toggler-icon",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="navbar-nav ml-auto"
            >
              <li
                className="nav-item active"
              >
                <Link
                  className="nav-link active"
                  exact="true"
                  replace={false}
                  to="/centers"
                >
                  Explore Our Centers
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link active"
                  replace={false}
                  to="/login"
                >
                  Sign In
                </Link>
              </li>
              <li
                className="nav-item dropdown"
                id="username_link"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  className="nav-link dropdown-toggle"
                  data-toggle="dropdown"
                  href="/"
                  id="navbarDropdownMenuLink"
                />
                <div
                  aria-labelledby="navbarDropdownMenuLink"
                  className="dropdown-menu"
                  id="username_sign_out"
                >
                  <Link
                    className="dropdown-item"
                    onClick={[MockFunction]}
                    replace={false}
                    to="/login"
                  >
                    Sign Out
                  </Link>
                </div>
              </li>
            </ul>,
            "className": "collapse navbar-collapse",
            "id": "navbarCollapse",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nav-item active"
                >
                  <Link
                    className="nav-link active"
                    exact="true"
                    replace={false}
                    to="/centers"
                  >
                    Explore Our Centers
                  </Link>
                </li>,
                undefined,
                undefined,
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link active"
                    replace={false}
                    to="/login"
                  >
                    Sign In
                  </Link>
                </li>,
                <li
                  className="nav-item dropdown"
                  id="username_link"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    className="nav-link dropdown-toggle"
                    data-toggle="dropdown"
                    href="/"
                    id="navbarDropdownMenuLink"
                  />
                  <div
                    aria-labelledby="navbarDropdownMenuLink"
                    className="dropdown-menu"
                    id="username_sign_out"
                  >
                    <Link
                      className="dropdown-item"
                      onClick={[MockFunction]}
                      replace={false}
                      to="/login"
                    >
                      Sign Out
                    </Link>
                  </div>
                </li>,
              ],
              "className": "navbar-nav ml-auto",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link active"
                    exact="true"
                    replace={false}
                    to="/centers"
                  >
                    Explore Our Centers
                  </Link>,
                  "className": "nav-item active",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Explore Our Centers",
                    "className": "nav-link active",
                    "exact": "true",
                    "replace": false,
                    "to": "/centers",
                  },
                  "ref": null,
                  "rendered": "Explore Our Centers",
                  "type": [Function],
                },
                "type": "li",
              },
              undefined,
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Link
                    className="nav-link active"
                    replace={false}
                    to="/login"
                  >
                    Sign In
                  </Link>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign In",
                    "className": "nav-link active",
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Sign In",
                  "type": [Function],
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a
                      aria-expanded="false"
                      aria-haspopup="true"
                      className="nav-link dropdown-toggle"
                      data-toggle="dropdown"
                      href="/"
                      id="navbarDropdownMenuLink"
                    />,
                    <div
                      aria-labelledby="navbarDropdownMenuLink"
                      className="dropdown-menu"
                      id="username_sign_out"
                    >
                      <Link
                        className="dropdown-item"
                        onClick={[MockFunction]}
                        replace={false}
                        to="/login"
                      >
                        Sign Out
                      </Link>
                    </div>,
                  ],
                  "className": "nav-item dropdown",
                  "id": "username_link",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-expanded": "false",
                      "aria-haspopup": "true",
                      "children": undefined,
                      "className": "nav-link dropdown-toggle",
                      "data-toggle": "dropdown",
                      "href": "/",
                      "id": "navbarDropdownMenuLink",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "a",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "aria-labelledby": "navbarDropdownMenuLink",
                      "children": <Link
                        className="dropdown-item"
                        onClick={[MockFunction]}
                        replace={false}
                        to="/login"
                      >
                        Sign Out
                      </Link>,
                      "className": "dropdown-menu",
                      "id": "username_sign_out",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Sign Out",
                        "className": "dropdown-item",
                        "onClick": [MockFunction],
                        "replace": false,
                        "to": "/login",
                      },
                      "ref": null,
                      "rendered": "Sign Out",
                      "type": [Function],
                    },
                    "type": "div",
                  },
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <a
            className="navbar-brand"
            href="/"
          >
            EventCity
          </a>
          <button
            aria-controls="navbarCollapse"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler navbar-toggler-right"
            data-target="#navbarCollapse"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>
          <div
            className="collapse navbar-collapse"
            id="navbarCollapse"
          >
            <ul
              className="navbar-nav ml-auto"
            >
              <li
                className="nav-item active"
              >
                <Link
                  className="nav-link active"
                  exact="true"
                  replace={false}
                  to="/centers"
                >
                  Explore Our Centers
                </Link>
              </li>
              <li
                className="nav-item"
              >
                <Link
                  className="nav-link active"
                  replace={false}
                  to="/login"
                >
                  Sign In
                </Link>
              </li>
              <li
                className="nav-item dropdown"
                id="username_link"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  className="nav-link dropdown-toggle"
                  data-toggle="dropdown"
                  href="/"
                  id="navbarDropdownMenuLink"
                />
                <div
                  aria-labelledby="navbarDropdownMenuLink"
                  className="dropdown-menu"
                  id="username_sign_out"
                >
                  <Link
                    className="dropdown-item"
                    onClick={[MockFunction]}
                    replace={false}
                    to="/login"
                  >
                    Sign Out
                  </Link>
                </div>
              </li>
            </ul>
          </div>
        </div>,
        "className": "navbar navbar-expand-md navbar-dark fixed-top bg-dark",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="navbar-brand"
              href="/"
            >
              EventCity
            </a>,
            <button
              aria-controls="navbarCollapse"
              aria-expanded="false"
              aria-label="Toggle navigation"
              className="navbar-toggler navbar-toggler-right"
              data-target="#navbarCollapse"
              data-toggle="collapse"
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>,
            <div
              className="collapse navbar-collapse"
              id="navbarCollapse"
            >
              <ul
                className="navbar-nav ml-auto"
              >
                <li
                  className="nav-item active"
                >
                  <Link
                    className="nav-link active"
                    exact="true"
                    replace={false}
                    to="/centers"
                  >
                    Explore Our Centers
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link active"
                    replace={false}
                    to="/login"
                  >
                    Sign In
                  </Link>
                </li>
                <li
                  className="nav-item dropdown"
                  id="username_link"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    className="nav-link dropdown-toggle"
                    data-toggle="dropdown"
                    href="/"
                    id="navbarDropdownMenuLink"
                  />
                  <div
                    aria-labelledby="navbarDropdownMenuLink"
                    className="dropdown-menu"
                    id="username_sign_out"
                  >
                    <Link
                      className="dropdown-item"
                      onClick={[MockFunction]}
                      replace={false}
                      to="/login"
                    >
                      Sign Out
                    </Link>
                  </div>
                </li>
              </ul>
            </div>,
          ],
          "className": "container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "EventCity",
              "className": "navbar-brand",
              "href": "/",
            },
            "ref": null,
            "rendered": "EventCity",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-controls": "navbarCollapse",
              "aria-expanded": "false",
              "aria-label": "Toggle navigation",
              "children": <span
                className="navbar-toggler-icon"
              />,
              "className": "navbar-toggler navbar-toggler-right",
              "data-target": "#navbarCollapse",
              "data-toggle": "collapse",
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "navbar-toggler-icon",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <ul
                className="navbar-nav ml-auto"
              >
                <li
                  className="nav-item active"
                >
                  <Link
                    className="nav-link active"
                    exact="true"
                    replace={false}
                    to="/centers"
                  >
                    Explore Our Centers
                  </Link>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link active"
                    replace={false}
                    to="/login"
                  >
                    Sign In
                  </Link>
                </li>
                <li
                  className="nav-item dropdown"
                  id="username_link"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    className="nav-link dropdown-toggle"
                    data-toggle="dropdown"
                    href="/"
                    id="navbarDropdownMenuLink"
                  />
                  <div
                    aria-labelledby="navbarDropdownMenuLink"
                    className="dropdown-menu"
                    id="username_sign_out"
                  >
                    <Link
                      className="dropdown-item"
                      onClick={[MockFunction]}
                      replace={false}
                      to="/login"
                    >
                      Sign Out
                    </Link>
                  </div>
                </li>
              </ul>,
              "className": "collapse navbar-collapse",
              "id": "navbarCollapse",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li
                    className="nav-item active"
                  >
                    <Link
                      className="nav-link active"
                      exact="true"
                      replace={false}
                      to="/centers"
                    >
                      Explore Our Centers
                    </Link>
                  </li>,
                  undefined,
                  undefined,
                  <li
                    className="nav-item"
                  >
                    <Link
                      className="nav-link active"
                      replace={false}
                      to="/login"
                    >
                      Sign In
                    </Link>
                  </li>,
                  <li
                    className="nav-item dropdown"
                    id="username_link"
                  >
                    <a
                      aria-expanded="false"
                      aria-haspopup="true"
                      className="nav-link dropdown-toggle"
                      data-toggle="dropdown"
                      href="/"
                      id="navbarDropdownMenuLink"
                    />
                    <div
                      aria-labelledby="navbarDropdownMenuLink"
                      className="dropdown-menu"
                      id="username_sign_out"
                    >
                      <Link
                        className="dropdown-item"
                        onClick={[MockFunction]}
                        replace={false}
                        to="/login"
                      >
                        Sign Out
                      </Link>
                    </div>
                  </li>,
                ],
                "className": "navbar-nav ml-auto",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-link active"
                      exact="true"
                      replace={false}
                      to="/centers"
                    >
                      Explore Our Centers
                    </Link>,
                    "className": "nav-item active",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Explore Our Centers",
                      "className": "nav-link active",
                      "exact": "true",
                      "replace": false,
                      "to": "/centers",
                    },
                    "ref": null,
                    "rendered": "Explore Our Centers",
                    "type": [Function],
                  },
                  "type": "li",
                },
                undefined,
                undefined,
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Link
                      className="nav-link active"
                      replace={false}
                      to="/login"
                    >
                      Sign In
                    </Link>,
                    "className": "nav-item",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign In",
                      "className": "nav-link active",
                      "replace": false,
                      "to": "/login",
                    },
                    "ref": null,
                    "rendered": "Sign In",
                    "type": [Function],
                  },
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <a
                        aria-expanded="false"
                        aria-haspopup="true"
                        className="nav-link dropdown-toggle"
                        data-toggle="dropdown"
                        href="/"
                        id="navbarDropdownMenuLink"
                      />,
                      <div
                        aria-labelledby="navbarDropdownMenuLink"
                        className="dropdown-menu"
                        id="username_sign_out"
                      >
                        <Link
                          className="dropdown-item"
                          onClick={[MockFunction]}
                          replace={false}
                          to="/login"
                        >
                          Sign Out
                        </Link>
                      </div>,
                    ],
                    "className": "nav-item dropdown",
                    "id": "username_link",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-expanded": "false",
                        "aria-haspopup": "true",
                        "children": undefined,
                        "className": "nav-link dropdown-toggle",
                        "data-toggle": "dropdown",
                        "href": "/",
                        "id": "navbarDropdownMenuLink",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "a",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "aria-labelledby": "navbarDropdownMenuLink",
                        "children": <Link
                          className="dropdown-item"
                          onClick={[MockFunction]}
                          replace={false}
                          to="/login"
                        >
                          Sign Out
                        </Link>,
                        "className": "dropdown-menu",
                        "id": "username_sign_out",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Sign Out",
                          "className": "dropdown-item",
                          "onClick": [MockFunction],
                          "replace": false,
                          "to": "/login",
                        },
                        "ref": null,
                        "rendered": "Sign Out",
                        "type": [Function],
                      },
                      "type": "div",
                    },
                  ],
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
